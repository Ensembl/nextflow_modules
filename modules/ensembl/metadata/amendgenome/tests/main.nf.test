// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test metadata/amendgenome
nextflow_process {

    name "Test Process METADATA_AMENDGENOME"
    script "../main.nf"
    process "METADATA_AMENDGENOME"

    tag "modules"
    tag "modules_ensembl"
    tag "metadata"
    tag "metadata/amendgenome"

    test("sarscov2 - bam - stub") {

        options "-stub"

        when {
            process {
                """
                input[0]= [
                    [id:'test'], 
                    file('${projectDir}/tests/modules/ensembl/seqregion/processseqregions/genome.json', checkIfExists: true),
                    file('${projectDir}/tests/modules/ensembl/seqregion/processseqregions/assembly_report.txt', checkIfExists: true),
                    file('${projectDir}/tests/modules/ensembl/seqregion/processseqregions/genome.fasta', checkIfExists: true),
                    file('${projectDir}/tests/modules/ensembl/seqregion/processseqregions/genome.gbff', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}