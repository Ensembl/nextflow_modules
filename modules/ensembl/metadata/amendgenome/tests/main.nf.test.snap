{
    "amendgenome (genome.json) - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "genome.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ce556b280384c5a933fb3005c92dba23"
                ],
                "amended_json": [
                    [
                        {
                            "id": "test"
                        },
                        "genome.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ce556b280384c5a933fb3005c92dba23"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-28T14:41:37.455102"
    }
}